pkg/operator/resources/utils/common.go

```diff
@@ -87,13 +87,17 @@ func CreatePortsContainer(name, image, pullPolicy string, ports []corev1.Contain
 }
 
 // CreateDeployment creates deployment
-func CreateDeployment(name, matchKey, matchValue, serviceAccountName string, replicas int32, infraNodePlacement *sdkapi.NodePlacement) *appsv1.Deployment {
+func CreateDeployment(name, matchKey, matchValue, serviceAccountName string, infraNodePlacement *cdiv1.InfraNodePlacement) *appsv1.Deployment {
 	podSpec := corev1.PodSpec{
 		SecurityContext: &corev1.PodSecurityContext{
 			RunAsNonRoot: &[]bool{true}[0],
 		},
 	}
-	deployment := ResourceBuilder.CreateDeployment(name, "", matchKey, matchValue, serviceAccountName, replicas, podSpec, infraNodePlacement)
+	var replicas int32 = 1
+	if infraNodePlacement != nil && infraNodePlacement.Replicas != nil {
```

Comment:
Here we check if `infraNodePlacement` is nil, but then we dereference it anyway. Maybe we could handle this a little bit better.

Reply:
Nice catch, thank you!

fixed in f62db8d79d46688bb5e958373e6c8bb8cb2dbeae

Reply:
Since maybe this comment got ignored, I will refer to it again:  https://github.com/kubevirt/containerized-data-importer/pull/2563#discussion_r1095156200

I think `infraNodePlacement.NodePlacement` should not be a pointer